python/openassetio_traitgen/__init__.py

```diff
@@ -136,27 +135,34 @@ def generate(
     # As a convenience, log the parsed structure
     _log_package_declaration(package_declaration, logger)
 
-    if dry_run:
+    # Don't allow empty string to fall through as getattr will throw.
+    if not generator:
+        logger.error("No generator specified")
+        return
+
+    # Retrieve the generator by looking up an attribute with the
+    # requested name and generate.
+    generator_obj = getattr(generators, generator, None)
+    if generator_obj is None:
+        logger.error("Cannot find generator %s", generator)
```

Comment:
[consider] Not sure on doing the argument validation here (sets a precedent for the other arguments). If we do I would lean toward an exception rather than log so the error can be communicated outside of this function.

Reply:
I'd be tempted to replace all this with  
```suggestion
    # Retrieve the generator by looking up an attribute with the
    # requested name and generate.
    generator_obj = getattr(generators, generator)
```
(i.e. without the `None` default) so that an `AttributeError` is raised if bad arguments are given.

Also see comment on requiring the generator command-line argument.

Reply:
Hmm, fair. I thought logging would be friendlier to the CLI, but to be fair I did make that decision before I understood you could limit the string inputs coming from there. 

I think we'll want to throw a better error than the `AttributeError` though, it's not very clear what goes wrong. I'll go with a try/catch/rethrow a better error sort of thing.

Reply:
[6c36bee](https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/7/commits/6c36bee3d2a7e98585a7ccfa454d6ce92a5d5acd)